### PR TITLE
bp: Add some more values to error messages

### PR DESCRIPTION
**Pull request by spencer-lunarg**
small part 2 of https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8470

### CODE REVIEW

File: layers/best_practices/bp_device_memory.cpp [hunk 1]
```diff
@@ -43,7 +43,8 @@ bool BestPractices::PreCallValidateAllocateMemory(VkDevice device, const VkMemor
 
     if ((Count<vvl::DeviceMemory>() + 1) > kMemoryObjectWarningLimit) {
         skip |= LogPerformanceWarning("BestPractices-vkAllocateMemory-too-many-objects", device, error_obj.location,
-                                      "This app has over %" PRIu32 " memory objects.", kMemoryObjectWarningLimit);
+                                      "This app has %zu memory objects, recommanded max is %" PRIu32 ".",
```
Comment (ziga-lunarg) [inline layers/best_practices/bp_device_memory.cpp:46]: Should `zu` be `PRIu64`?

File: layers/best_practices/bp_synchronization.cpp [hunk 2]
```diff
@@ -350,14 +350,17 @@ bool BestPractices::PreCallValidateCmdPipelineBarrier(
...
+        const uint32_t num = num_barriers_objects_.load();
+        const uint32_t total_barriers = num + imageMemoryBarrierCount + bufferMemoryBarrierCount;
+        if (total_barriers > kMaxRecommendedBarriersSizeAMD) {
             skip |= LogPerformanceWarning("BestPractices-AMD-CmdBuffer-highBarrierCount", commandBuffer, error_obj.location,
-                                          "%s In this frame, %" PRIu32
-                                          " barriers were already submitted. Barriers have a high cost and can "
+                                          "%s In this frame, %" PRIu32 " barriers were already submitted (%" PRIu32
+                                          " if you include image and buffer barriers too). Barriers have a high cost and can "
                                           "stall the GPU. "
+                                          "Total recommanded max is %" PRIu32
```
Comment (ziga-lunarg) [inline layers/best_practices/bp_synchronization.cpp:360]: `recommended`

File: layers/best_practices/bp_synchronization.cpp [hunk 3]
```diff
@@ -524,7 +527,7 @@ bool BestPractices::PreCallValidateCreateSemaphore(VkDevice device, const VkSema
                                           "%zu created, but recommanded max is %" PRIu32
                                           ". "
                                           "Minimize the amount of queue synchronization that is used. "
-                                          "Semaphores and fences have overhead. Each fence has a CPU and GPU cost with it.",
+                                          "Semaphores and fences have overhead. Each semaphores has a CPU and GPU cost with it.",
```
Comment (ziga-lunarg) [inline layers/best_practices/bp_synchronization.cpp:530]: `Each semaphores has` is not making much sense, did you mean `Each fence and semaphore has`

Comment (spencer-lunarg) [reply layers/best_practices/bp_synchronization.cpp:530]: there are 2 checks, 1 for fence and 1 for semaphore... will just remove the mention of the other object out of these messages

File: layers/best_practices/bp_synchronization.cpp [hunk 4]
```diff
@@ -521,10 +524,10 @@ bool BestPractices::PreCallValidateCreateSemaphore(VkDevice device, const VkSema
         if (count > kMaxRecommendedSemaphoreObjectsSizeAMD) {
             skip |= LogPerformanceWarning("BestPractices-SyncObjects-HighNumberOfSemaphores", device, error_obj.location,
                                           "%s %s High number of vkSemaphore objects created. "
-                                          "%zu created, but recommanded max is %" PRIu32
+                                          "%zu created, but recommended max is %" PRIu32
                                           ". "
                                           "Minimize the amount of queue synchronization that is used. "
-                                          "Semaphores and fences have overhead. Each fence has a CPU and GPU cost with it.",
+                                          "Each semaphores has a CPU and GPU overhead cost with it.",
```
Comment (ziga-lunarg) [inline layers/best_practices/bp_synchronization.cpp:530]: `Each semaphore`

File: layers/best_practices/bp_device_memory.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply layers/best_practices/bp_device_memory.cpp]: no, made it explicit, but it is a `size_t` and the iOS build got mad

**Review by ziga-lunarg — APPROVED** [review]

**Pull request merged**